ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/rcf/BatchRandomCutForest.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.ml.engine.algorithms.rcf;
+
+import com.amazon.randomcutforest.RandomCutForest;
+import com.amazon.randomcutforest.state.RandomCutForestMapper;
+import com.amazon.randomcutforest.state.RandomCutForestState;
+import lombok.extern.log4j.Log4j2;
+import org.opensearch.ml.common.dataframe.ColumnMeta;
+import org.opensearch.ml.common.dataframe.ColumnValue;
+import org.opensearch.ml.common.dataframe.DataFrame;
+import org.opensearch.ml.common.dataframe.DataFrameBuilder;
+import org.opensearch.ml.common.dataframe.Row;
+import org.opensearch.ml.common.parameter.BatchRCFParams;
+import org.opensearch.ml.common.parameter.FunctionName;
+import org.opensearch.ml.common.parameter.MLAlgoParams;
+import org.opensearch.ml.common.parameter.MLOutput;
+import org.opensearch.ml.common.parameter.MLPredictionOutput;
+import org.opensearch.ml.common.parameter.Model;
+import org.opensearch.ml.engine.TrainAndPredictable;
+import org.opensearch.ml.engine.annotation.Function;
+import org.opensearch.ml.engine.utils.ModelSerDeSer;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * Use RCF to detect non-time-series data.
+ */
+@Log4j2
+@Function(FunctionName.BATCH_RCF)
+public class BatchRandomCutForest implements TrainAndPredictable {
+    private static final int DEFAULT_NUMBER_OF_TREES = 30;
```

Comment:
curiosity question: do these values have to be aligned with fixed in time rcf?

Reply:
No. They could be different.